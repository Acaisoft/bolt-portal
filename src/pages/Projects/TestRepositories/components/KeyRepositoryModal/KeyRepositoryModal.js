import React from 'react'
import Button from '@material-ui/core/Button'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogTitle from '@material-ui/core/DialogTitle'
import { withStyles } from '@material-ui/core/styles'

import styles from './KeyRepositoryModal.styles'

export class KeyRepositoryModal extends React.Component {
  render() {
    return (
      <div>
        <Dialog
          open={this.props.open}
          onClose={this.props.handleClose}
          aria-labelledby="modal-key-title"
          aria-describedby="modal-key-description"
        >
          <DialogTitle id="modal-key-title">{'REPOSITORY KEY'}</DialogTitle>
          <DialogContent>
            <DialogContentText id="modal-key-description">
              Copy it and do the magic in your repository: <br />
              Generated Key(TODO)...
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button
              onClick={this.props.handleClose}
              color="primary"
              variant="contained"
              autoFocus
            >
              OK
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    )
  }
}

export default withStyles(styles)(KeyRepositoryModal)
