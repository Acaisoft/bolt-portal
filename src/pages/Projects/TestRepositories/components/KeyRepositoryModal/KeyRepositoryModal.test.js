import React from 'react'
import { shallow } from 'enzyme'

import { KeyRepositoryModal } from './KeyRepositoryModal'

import { mocks } from '~utils/tests'

const initComponent = overrides => {
  const mockProps = {
    open: false,
    close: jest.fn(),
  }
  const wrapper = shallow(
    <KeyRepositoryModal {...mockProps} {...overrides} classes={mocks.ClassesProxy} />
  )
  const instance = wrapper.instance()
  return { instance, mockProps, wrapper }
}

describe('component: KeyRepositoryModal', () => {
  describe('rendering', () => {
    it('should render without crashing', () => {
      const { wrapper } = initComponent()
      expect(wrapper).toBeTruthy()
    })
    it('should render as expected', () => {
      const { wrapper } = initComponent()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
