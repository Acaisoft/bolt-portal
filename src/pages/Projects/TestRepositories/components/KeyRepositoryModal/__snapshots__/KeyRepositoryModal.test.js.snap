// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: KeyRepositoryModal rendering should render as expected 1`] = `
<div>
  <WithStyles(Dialog)
    aria-describedby="modal-key-description"
    aria-labelledby="modal-key-title"
    open={false}
  >
    <WithStyles(DialogTitle)
      id="modal-key-title"
    >
      REPOSITORY KEY
    </WithStyles(DialogTitle)>
    <WithStyles(DialogContent)>
      <WithStyles(DialogContentText)
        id="modal-key-description"
      >
        Copy it and do the magic in your repository: 
        <br />
        Generated Key(TODO)...
      </WithStyles(DialogContentText)>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        autoFocus={true}
        color="primary"
        variant="contained"
      >
        OK
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</div>
`;
